private/pkg/observabilityzap/tracer_provider_closer.go

```diff
@@ -20,12 +20,15 @@ import (
 
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/embedded"
 )
 
 var _ trace.TracerProvider = &tracerProviderCloser{}
 var _ io.Closer = &tracerProviderCloser{}
 
 type tracerProviderCloser struct {
+	// https://pkg.go.dev/go.opentelemetry.io/otel/trace#hdr-API_Implementations
```

Comment:
Otel is the strangest package. How is this interface implemented? It contains an unexported method, so my understanding is that it can _never_ be implemented outside of `otel/trace/embedded`.

Reply:
I agree - pushed a follow up which will get us out of the business of implementing our own `TracerProvider` and instead just return a separate `io.Closer` to allow shutting down the tracer provider.